### PR TITLE
Rename integration Windows ATP to Microsoft ATP

### PR DESCRIPTION
**Pull request by teizenman**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/17297
fixes: https://github.com/demisto/etc/issues/20461

## Description
Microsoft-atp is a V2 integration for windows-atp, added new commands, and re-write the old ones

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by Itay4** [comment]
what are we trying to do here? this breaks bc..

**Comment by barchen1** [comment]
@Itay4 
I talked with @Arsenikr who said we need to rename this integration from Windows ATP to Microsoft ATP. 
@Arsenikr - Where did this decision originate ?

**Comment by Itay4** [comment]
lets continue discussion at https://github.com/demisto/etc/issues/17297

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 1]
```diff
@@ -0,0 +1,1778 @@
...
+    return hr, ec, response
+
+
+def fetch_incidents():
+    last_run = demisto.getLastRun()
+
+    if last_run and last_run['last_alert_fetched_time']:
+        last_alert_fetched_time = datetime.strptime(last_run['last_alert_fetched_time'], '%Y-%m-%dT%H:%M:%S.%f')
+    else:
+        last_alert_fetched_time = datetime.now() - timedelta(days=300)
```
Comment (guykeller) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: outer function

Comment (guykeller) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: 300 should be a const, and should probably be a parameter and not a magic number

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 2]
```diff
@@ -0,0 +1,1778 @@
...
+    incidents = []
+    last_ids = []
+
+    for alert in alerts:
+        # Removing 'Z' from timestamp and converting to datetime
+        alert_creation_time = datetime.strptime(alert['alertCreationTime'][:-2], '%Y-%m-%dT%H:%M:%S.%f')
+        alert_status = alert['status']
+        alert_severity = alert['severity']
+        if alert_creation_time >= last_alert_fetched_time and alert_status in FETCH_STATUS and \
+                alert_severity in FETCH_SEVERITY and alert['id'] not in previous_ids:
```
Comment (guykeller) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: What are the two middle conditions?
current names make them unclear

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 3]
```diff
@@ -0,0 +1,1778 @@
...
+    response = requests.get(
+        BASE_URL + '/alerts',
+        headers={
+            'Authorization': 'Bearer ' + token,
+            'Content-Type': 'application/json',
+            'Accept': 'application/json'
+        },
+        params={'$top': '1'},
+        verify=USE_SSL
+    )
```
Comment (guykeller) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: what request is this?

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 4]
```diff
@@ -0,0 +1,1778 @@
...
+    try:
+        _ = response.json() if response.text else {}
+        if not response.ok:
+            return_error(f'API call to Windows Advanced Threat Protection. '
+                         f'Please check authentication related parameters. '
+                         f'[{response.status_code}] - {response.reason}')
+
```
Comment (guykeller) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: this error message should be improved

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml [hunk 5]
```diff
@@ -0,0 +1,2221 @@
+category: Endpoint
+commonfields:
```
Comment (guykeller) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml]: Could you please specify here which changes you made to which commands,  and which commands did you add from scratch?

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection_test.py [hunk 6]
```diff
@@ -0,0 +1,541 @@
...
+    import MicrosoftDefenderAdvancedThreatProtection as atp
+    mocker.patch.object(atp, 'stop_and_quarantine_file_request', return_value=MACHINE_ACTION_API_RESPONSE)
+    mocker.patch.object(atp, 'get_machine_action_data', return_value=MACHINE_ACTION_DATA)
+    _, res, _ = atp.stop_and_quarantine_file_command()
+    assert res['MicrosoftATP.MachineAction(val.ID === obj.ID)'] == MACHINE_ACTION_DATA
+
+
+def test_get_investigations_by_id_command(mocker):
+    import MicrosoftDefenderAdvancedThreatProtection as atp
+    mocker.patch.object(demisto, 'args', return_value={'id': '123', 'limit': '1', 'offset': '0'})
```
Comment (guykeller) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection_test.py]: add tests for negative /  non int  limit and offset, here and for every other command

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection_test.py [hunk 7]
```diff
@@ -0,0 +1,541 @@
...
+
+
+def atp_mocker(mocker):
+    import MicrosoftDefenderAdvancedThreatProtection as atp
+    with open('test_data/alerts.json', 'r') as f:
+        alerts = json.loads(f.read())
+    mocker.patch.object(atp, 'list_alerts_request', return_value=alerts)
+
+
+def test_fetch(mocker):
```
Comment (guykeller) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection_test.py]: by moving functionality in fetch to outer functions, you'll be able to thoroughly test each part of the functionality separately.
this ability is extremely important in fetch, which is very bug prone.
please add significantly more tests to it after you export the functions - test that the next run values are updated  correctly, that the limit works as expected,  test every assumption that you have about what works in this function

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Playbooks/Microsoft_Defender_Advanced_Threat_Protection_Get_machine_action_status.yml [hunk 8]
```diff
@@ -0,0 +1,171 @@
+elasticcommonfields: {}
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Playbooks/Microsoft_Defender_Advanced_Threat_Protection_Get_machine_action_status.yml]: add a screenshot of the PB to the comments in the PR please

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml [hunk 9]
```diff
@@ -0,0 +1,2249 @@
...
+  name: fetch_status
+  required: false
+  type: 0
+- defaultvalue: Informational,Low,Medium,High
+  display: 'Severity to filter out alerts for fetching as incidents. The property
+    values are: Informational,Low,Medium,High (Comma separated values supported, e.g.
+    Medium,High)'
+  name: fetch_severity
+  required: false
+  type: 0
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml]: make them both multi select

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 10]
```diff
@@ -0,0 +1,1941 @@
...
+import requests
+import base64
+from cryptography.hazmat.primitives.ciphers.aead import AESGCM
+requests.packages.urllib3.disable_warnings()
+
+if not demisto.params()['proxy']:
+    del os.environ['HTTP_PROXY']
+    del os.environ['HTTPS_PROXY']
+    del os.environ['http_proxy']
+    del os.environ['https_proxy']
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: use handle_proxy() and move this to the try/except block

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 11]
```diff
@@ -0,0 +1,1941 @@
...
+if len(AUTH_AND_TOKEN_URL) != 2:
+    TOKEN_RETRIEVAL_URL = 'https://oproxy.demisto.ninja/obtain-token'  # disable-secrets-detection
+else:
+    TOKEN_RETRIEVAL_URL = AUTH_AND_TOKEN_URL[1]
+APP_NAME = 'ms-defender-atp'
+
+''' HELPER FUNCTIONS '''
+
+
+def epoch_seconds(d=None):
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: @roysagi don't we have something like this in CommonServerPython?

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 12]
```diff
@@ -0,0 +1,1941 @@
...
+        return r.json()
+    except ValueError:
+        return {}
+
+
+def alert_to_incident(alert, alert_creation_time):
+    incident = {}
+    incident['rawJSON'] = json.dumps(alert)
+    incident['name'] = 'Microsoft Defender ATP Alert ' + alert['id']
+    incident['occurred'] = alert_creation_time.isoformat() + 'Z'
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]:
```suggestion
    incident = {
        'rawJSON': json.dumps(alert)
        'name': 'Microsoft Defender ATP Alert ' + alert['id']
        'occurred': alert_creation_time.isoformat() + 'Z'
    }
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 13]
```diff
@@ -0,0 +1,1941 @@
...
+
+    return incident
+
+
+def capitalize_first_letter(string):
+    return string[:1].upper() + string[1:]
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: dead code, please remove

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 14]
```diff
@@ -0,0 +1,1941 @@
...
+
+
+''' REQUESTS '''
+
+
+def isolate_machine_request(machine_id, comment, isolation_type):
+    """Isolates a machine from accessing external network..
+    Args:
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: add a blank line before each section

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 15]
```diff
@@ -0,0 +1,1941 @@
...
+
+def capitalize_first_letter(string):
+    return string[:1].upper() + string[1:]
+
+
+''' REQUESTS '''
+
+
+def isolate_machine_request(machine_id, comment, isolation_type):
+    """Isolates a machine from accessing external network..
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]:
```suggestion
    """Isolates a machine from accessing external network.
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 16]
```diff
@@ -0,0 +1,1941 @@
...
+    Args:
+        machine_id (str): Machine ID
+        comment (str): Comment to associate with the action.
+        isolation_type (str): Type of the isolation.
+    Returns:
+        # Machine action is a collection of actions you can apply on the machine, for more info
+         https://docs.microsoft.com/en-us/windows/security/threat-protection/microsoft-defender-atp/machineaction
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: this should be in a note section

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 17]
```diff
@@ -0,0 +1,1778 @@
...
+def isolate_machine_command():
+    """Isolates a machine from accessing external network.
+    Returns:
+        (str, dict, dict). Human readable, context, raw response
+    """
+    machine_id = demisto.args().get('machine_id')
+    comment = demisto.args().get('comment')
+    isolation_type = demisto.args().get('isolation_type')
+    response = isolate_machine_request(machine_id, comment, isolation_type)
+    context_output = get_machine_action_data(response['id'])
```
[the remark this answers is not in the document]
Comment (ronykoz) [reply Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: not sure that i follow, if i want to just isolate the machine why do i need to get the machine_action data?

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 18]
```diff
@@ -0,0 +1,1778 @@
...
+        (str, dict, dict). Human readable, context, raw response
+    """
+    headers = ['ID', 'ComputerDNSName', 'OSPlatform', 'LastIPAddress', 'LastExternalIPAddress', 'HealthStatus',
+               'RiskScore', 'ExposureLevel']
+    hostname = demisto.args().get('hostname', '')
+    ip = demisto.args().get('ip', '')
+    risk_score = demisto.args().get('risk_score', '')
+    health_status = demisto.args().get('health_status', '')
+    os_platform = demisto.args().get('os_platform', '')
+    filter_fields_dict = {'computerDnsName': hostname, 'lastIpAddress': ip, 'riskScore': risk_score,
```
[the remark this answers is not in the document]
Comment (ronykoz) [reply Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]:
```suggestion
    fields_to_filter_by = {'computerDnsName': hostname, 'lastIpAddress': ip, 'riskScore': risk_score,
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 19]
```diff
@@ -0,0 +1,1941 @@
...
+    """
+    headers = ['ID', 'ComputerDNSName', 'OSPlatform', 'LastIPAddress', 'LastExternalIPAddress', 'HealthStatus',
+               'RiskScore', 'ExposureLevel']
+    hostname = demisto.args().get('hostname', '')
+    ip = demisto.args().get('ip', '')
+    risk_score = demisto.args().get('risk_score', '')
+    health_status = demisto.args().get('health_status', '')
+    os_platform = demisto.args().get('os_platform', '')
+    # A dictionary that contains all of the fields the user want to filter results by.
+    # It will be sent in the request so the requested filters are applied on the results
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]:
```suggestion
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 20]
```diff
@@ -0,0 +1,1941 @@
...
+    filter_fields_dict = {'computerDnsName': hostname, 'lastIpAddress': ip, 'riskScore': risk_score,
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: Dicts in general should be multi line

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 21]
```diff
@@ -0,0 +1,1941 @@
...
+    """
+    machines_list = []
+    for machine in machines_response['value']:
+        machine_data = get_machine_data(machine['id'])
+        machines_list.append(machine_data)
+    return machines_list
+
+
+def get_filter_request(filter_fields_dict):
+    """Get a dict with all of the fields to filter
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: it's not a dict

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 22]
```diff
@@ -0,0 +1,1941 @@
...
+        list. Machines list
+    """
+    machines_list = []
+    for machine in machines_response['value']:
+        machine_data = get_machine_data(machine['id'])
+        machines_list.append(machine_data)
+    return machines_list
+
+
+def get_filter_request(filter_fields_dict):
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]:
```suggestion
def reformat_filter(filter_fields_dict):
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 23]
```diff
@@ -0,0 +1,1941 @@
...
+    risk_score = demisto.args().get('risk_score', '')
+    health_status = demisto.args().get('health_status', '')
+    os_platform = demisto.args().get('os_platform', '')
+    # A dictionary that contains all of the fields the user want to filter results by.
+    # It will be sent in the request so the requested filters are applied on the results
+    filter_fields_dict = {'computerDnsName': hostname, 'lastIpAddress': ip, 'riskScore': risk_score,
+                          'healthStatus': health_status, 'osPlatform': os_platform}
+    filter_req = get_filter_request(filter_fields_dict)
+    machines_response = get_machines_request(filter_req)
+    machines_list = get_machines_list(machines_response)
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: why are we executing so many queries

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 24]
```diff
@@ -0,0 +1,1941 @@
...
+               'RiskScore', 'ExposureLevel']
+    file = demisto.args()['file_hash']
+    machines_response = get_file_related_machines_request(file)
+    machines_list = get_machines_list(machines_response)
+    context_output = {
+        'File': file,
+        'Machines': machines_list
+    }
+    ec = {
+        'MicrosoftATP.FileMachine(val.ID === obj.ID)': context_output
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: FileMachine?

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 25]
```diff
@@ -0,0 +1,1941 @@
...
+def add_args_to_json_and_context(alert_id, assigned_to, status, classification, determination, comment):
+    """Gets arguments and returns the json and context with the arguments inside
+    """
+    json = {}
+    context = {
+        'ID': alert_id
+    }
+    if assigned_to:
+        json['assignedTo'] = assigned_to
+        context['AssignedTo'] = assigned_to
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: read about camelize in commonserverpython

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 26]
```diff
@@ -0,0 +1,1941 @@
...
+        files_data_list.append(get_file_data(file_obj['sha1']))
+    context_output = {
+        'AlertID': alert_id,
+        'Files': files_data_list
+    }
+    ec = {
+        'MicrosoftATP.AlertFile(val.AlertID === obj.AlertID)': context_output
+    }
+    hr = tableToMarkdown(f'Alert {alert_id} Related Files:', files_data_list, headers=headers)
+    return hr, ec, response_files_list
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: add blank lines to separate logic

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 27]
```diff
@@ -0,0 +1,1941 @@
...
+        return limit_int, offset_int
+
+
+def get_file_data(file_sha1):
+    """Get file's hash and returns the file's info.
+    Returns:
+        dict. File's info
+    """
+    response = get_file_data_request(file_sha1)
+    file_data = assign_params(**{
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: LIKE

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 28]
```diff
@@ -0,0 +1,1941 @@
...
+    offset = demisto.args().get('offset')
+    limit, offset = check_limit_and_offset_values(limit, offset)
+    response = get_alert_related_ips_request(alert_id)
+    response_ips_list = response['value']
+    ips_list = []
+    from_index = min(offset, len(response_ips_list))
+    to_index = min(offset + limit, len(response_ips_list))
+    for ip in response_ips_list[from_index:to_index]:
+        ips_list.append(ip['id'])
+    context_output = {
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: add blank line

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 29]
```diff
@@ -0,0 +1,1941 @@
...
+        filter_alerts_creation_time = filter_alerts_creation_time + "Z"
+
+    return filter_alerts_creation_time
+
+
+def all_alerts_to_incidents(alerts, latest_creation_time, existing_ids):
+    """Gets the alerts list and convert it to incidents
+    Args:
+        alerts(list): List of alerts filtered by the first_fetch_timestamp parameter
+        latest_creation_time(date):  La
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: WHY?

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 30]
```diff
@@ -0,0 +1,1941 @@
...
+    ec = {
+        'MicrosoftATP.Machine(val.ID === obj.ID)': machine_data
+    }
+    return hr, ec, response
+
+
+def fetch_incidents():
+    last_run = demisto.getLastRun()
+    last_alert_fetched_time = get_last_alert_fetched_time(last_run)
+    existing_ids = last_run.get('existing_ids', [])
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: why do we need this

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py (hunk 29 above)
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]:
```suggestion
            return_error(f'API call to Windows Advanced Threat Protection. '
```
```suggestion
            return_error(f'API call to Windows Advanced Threat Protection Failed. '
```

**Comment by liorblob** [comment]
For an example to use the API module, take a look at Microsoft Graph Mail/Single User

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py (hunk 29 above)
Comment (barchen1) [reply Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: changed timestamp from ' %Y-%m-%dT%H:%M:%S.%f' to '%Y-%m-%dT%H:%M:%S' as we talked

**Comment by barchen1** [comment]
Here is a screenshot of Microsoft Defender Advanced Threat Protection Get Machine Action Status
playbook
image

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml (hunk 9 above)
Comment (barchen1) [reply Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml]: done

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection_description.md [hunk 31]
```diff
@@ -0,0 +1,2 @@
+To allow us access to Windows Defender, an admin has to approve our app using an admin consent flow, by clicking on the following [link](https://oproxy.demisto.ninja/ms-defender-atp).
+After authorizing the Demisto app, you will get an ID, Token, and Key, which should be inserted in the integration instance configuration's corresponding fields.
```
[the remark this answers is not in the document]
Comment (barchen1) [reply Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection_description.md]: done

File: Integrations/WindowsDefenderAdvancedThreatProtection/CHANGELOG.md [hunk 32]
```diff
@@ -1,5 +1,5 @@
 ## [Unreleased]
-
+Use Microsoft Defender Advanced Threat Protection integration instead.
```
[the remark this answers is not in the document]
Comment (barchen1) [reply Integrations/WindowsDefenderAdvancedThreatProtection/CHANGELOG.md]: done

**Comment by ronykoz** [comment]
@kirbles19 to make sure that we don't get lost among the many non-related files here:
The relevant files are:
https://github.com/demisto/content/blob/renaming-windows-defender/Integrations/WindowsDefenderAdvancedThreatProtection/CHANGELOG.md
https://github.com/demisto/content/blob/renaming-windows-defender/Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml
https://github.com/demisto/content/blob/renaming-windows-defender/Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection_description.md
https://github.com/demisto/content/blob/renaming-windows-defender/Packs/MicrosoftDefenderAdvancedThreatProtection/Playbooks/Microsoft_Defender_Advanced_Threat_Protection_Get_Machine_Action_Status.yml

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 33]
```diff
@@ -0,0 +1,2111 @@
...
+    action_id = demisto.args().get('id', '')
+    status = demisto.args().get('status', '')
+    machine_id = demisto.args().get('machine_id', '')
+    type = demisto.args().get('type', '')
+    requestor = demisto.args().get('requestor', '')
+    if action_id:
+        for index in range(3):
+            try:
+                response = get_machine_action_by_id_request(action_id)
+                time.sleep(1)
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]:
```suggestion
                response = get_machine_action_by_id_request(action_id)
                if response:
                    break
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 34]
```diff
@@ -0,0 +1,2111 @@
...
+            except Exception as e:
+                if 'ResourceNotFound' in str(e) and index < 3:
+                    continue
```
Comment (ronykoz) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]:
```suggestion
                    time.sleep(1)
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py (hunk 33 above)
Comment (barchen1) [reply Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: done

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py (hunk 34 above)
Comment (barchen1) [reply Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py]: done

**Pull request closed without merging**